lib/matplotlib/axes/_base.py

```diff
@@ -1594,39 +1595,46 @@ def axis(self, *v, **kwargs):
 
         Parameters
         ----------
-        v : list of float or {'on', 'off', 'equal', 'tight', 'scaled',\
-            'normal', 'auto', 'image', 'square'}
-            Optional positional argument
-
-            Axis data limits set from a list; or a command relating to axes:
-
-                ========== ================================================
-                Value      Description
-                ========== ================================================
-                'on'       Toggle axis lines and labels on
-                'off'      Toggle axis lines and labels off
-                'equal'    Equal scaling by changing limits
-                'scaled'   Equal scaling by changing box dimensions
-                'tight'    Limits set such that all data is shown
-                'auto'     Automatic scaling, fill rectangle with data
-                'normal'   Same as 'auto'; deprecated
-                'image'    'scaled' with axis limits equal to data limits
-                'square'   Square plot; similar to 'scaled', but initially\
-                           forcing xmax-xmin = ymax-ymin
-                ========== ================================================
+        v : List[float] or one of the strings listed below.
+            Optional positional-only argument
+
+            If a list, set the axis data limits.  If a string:
+
+            ======== ==========================================================
+            Value    Description
+            ======== ==========================================================
+            'on'     Turn on axis lines and labels.
+            'off'    Turn off axis lines and labels.
+            'equal'  Set equal scaling (i.e., make circles circular) by
+                     changing axis limits.
+            'scaled' Set equal scaling (i.e., make circles circular) by
+                     changing dimensions of the plot box.
+            'tight'  Set limits such that all data is shown; if this is already
+                     the case, move data to the center of the figure without
+                     modifying ``xmax-xmin`` or ``ymax-ymin``.
```

Comment:
I don't understand; what is an example of the second case, where the limits are changed without changing their differences?

Reply:
Indeed the description seems to be incorrect.

~~~
>>> fig, ax = plt.subplots()
>>> ax.plot([1,3,2])
>>> ax.get_xlim()
(-0.1, 2.1)

>>> ax.set_xlim(right=4.1)
(-0.1, 4.1)

>>> ax.axis('tight')
(-0.1, 4.1, 0.9, 3.1)

>>> ax.get_xlim()
(-0.1, 2.1)
~~~

From the description I would have anticipated `(-1.1, 3.1)`. Anyway, the implemented behavior seems more reasonable. I don't see why 'tight' should preserve `xmax-xmin`. So the docstring should be changed.

Reply:
Well, I just copied the pyplot docstring :-) Removed the offending sentence, although now it's unclear how this is different from "auto".

As a side note (but not changed by this PR), the docstring of autoscale() states that
```
        tight: bool or None, optional
            If True, set view limits to data limits;
            if False, let the locator and margins expand the view limits;
            if None, use tight scaling if the only artist is an image,
            otherwise treat *tight* as False.
            The *tight* setting is retained for future autoscaling
            until it is explicitly changed.
```
implying that `autoscale=True` ignores margins.  But either that's not true, or `axis('tight')` has a different meaning per @timhoffm's example.  Note also the docstring of `autoscale_view`:
```
        If *tight* is *False*, the axis major locator will be used
        to expand the view limits if rcParams['axes.autolimit_mode']
        is 'round_numbers'.  Note that any margins that are in effect
        will be applied first, regardless of whether *tight* is
        *True* or *False*.  Specifying *tight* as *True* or *False*
        saves the setting as a private attribute of the Axes; specifying
        it as *None* (the default) applies the previously saved value.
```
so is `tight=False` basically redundant with the axes.autolimit_mode rc?  All that's very confusing...